core/arch/atomic_exports.h

```diff
@@ -723,7 +723,254 @@ atomic_dec_becomes_zero(volatile int *var)
+                ASSERT_CURIOSITY(!hot_patch);                        \
+                __asm__ __volatile__("fence iorw,iorw\n\t"           \
+                                     "sw %0, 0(%1)\n\t"              \
+                                     :                               \
+                                     : "r"(value), "r"(target)       \
+                                     : "memory");                    \
+            } while (0)
+#        define ATOMIC_4BYTE_ALIGNED_READ(addr_src, addr_res)       \
+            do {                                                    \
+                __asm__ __volatile__("lwu %0, 0(%1)\n\t"            \
+                                     "fence iorw,iorw"              \
+                                     :                              \
+                                     : "r"(addr_src), "r"(addr_res) \
+                                     : "memory");                   \
+            } while (0)
+#        define ATOMIC_8BYTE_WRITE(target, value, hot_patch)   \
+            do {                                               \
+                /* Assuming little endian. */                  \
+                ASSERT(BYTE_ORDER == LITTLE_ENDIAN);           \
+                /* FIXME i#3544: test the assembly!*/          \
+                __asm__ __volatile__("fence iorw,iorw\n\t"     \
+                                     "sb %0, 0(%1)\n\t"        \
+                                     "srli %0, %0, 8\n\t"      \
+                                     "sb %0, 1(%1)\n\t"        \
+                                     "srli %0, %0, 8\n\t"      \
+                                     "sb %0, 2(%1)\n\t"        \
+                                     "srli %0, %0, 8\n\t"      \
+                                     "sb %0, 3(%1)\n\t"        \
+                                     "srli %0, %0, 8\n\t"      \
+                                     "sb %0, 4(%1)\n\t"        \
+                                     "srli %0, %0, 8\n\t"      \
+                                     "sb %0, 5(%1)\n\t"        \
+                                     "srli %0, %0, 8\n\t"      \
+                                     "sb %0, 6(%1)\n\t"        \
+                                     "srli %0, %0, 8\n\t"      \
+                                     "sb %0, 7(%1)\n\t"        \
+                                     :                         \
+                                     : "r"(value), "r"(target) \
+                                     : "memory");              \
+            } while (0)
+#        define ATOMIC_8BYTE_ALIGNED_WRITE(target, value, hot_patch) \
+            do {                                                     \
+                /* Not currently used to write code */               \
+                ASSERT_CURIOSITY(!hot_patch);                        \
+                __asm__ __volatile__("fence iorw,iorw\n\t"           \
```

Comment:
Also assert that the given target is aligned. Same in other ALIGNED macros.

Reply:
Done. As I require aligned addresses for essentially all atomic operations, I've added asserts to each of them.